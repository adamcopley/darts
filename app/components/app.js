import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import {
    StyleSheet,
    Text,
    View,
    Button
} from 'react-native'

import ScoreBoard from './scoreboard'
import KeyBoard from './keyboard'

import {
    onRestartGame,
    onScoreClear
} from '../actions'

const mapStateToProps = state => ({
    error: state.error,
    p1: state.p1,
    p2: state.p2,
    currentScore: state.currentScore
})

const mapDispatchToProps = dispatch => bindActionCreators({
    onRestartGame,
    onScoreClear
}, dispatch)

class App extends Component {
    render() {
        let {
            p1,
            p2,
            currentScore,
            onScoreClear,
            onRestartGame,
            error
        } = this.props

        return (
            <View style={styles.container}>
                <ScoreBoard />
                <Text>Ð¡urrent turn {currentScore}</Text>
                <Text>
                    {error} {error && (<Text onPress={onScoreClear}>Clear score</Text>)}
                </Text>
                <KeyBoard />
                <Button
                    onPress={onRestartGame}
                    title="Restart"
                />
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    }
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
