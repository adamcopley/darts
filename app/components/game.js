import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import {
    StyleSheet,
    Text,
    View,
    Button
} from 'react-native'

import ScoreBoard from './scoreboard'
import KeyBoard from './keyboard'

import {
    onGameStartNew
} from '../actions'

const mapStateToProps = state => ({
    error: state.error,
    currentScore: state.currentScore
})

const mapDispatchToProps = dispatch => bindActionCreators({
    onGameStartNew
}, dispatch)

const Game = ({
    onGameStartNew,
    currentScore,
    error,
}) => (
    <View style={styles.container}>
        <ScoreBoard />
        <Text>Ð¡urrent turn {currentScore}</Text>
        <Text>{error}</Text>
        <KeyBoard />
        <Button
            onPress={onGameStartNew}
            title="Start New"
        />
    </View>
)

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    }
})

export default connect(mapStateToProps, mapDispatchToProps)(Game)
