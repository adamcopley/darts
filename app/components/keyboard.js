import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { StyleSheet, View, TouchableOpacity, Text } from 'react-native'

import {
    onScoreChange,
    onScoreSubmit,
    onScoreClear,
    onScoreUndo
} from '../actions'

let buttons = [
    [{title: '1'}, {title: '2'}, {title: '3'}],
    [{title: '4'}, {title: '5'}, {title: '6'}],
    [{title: '7'}, {title: '8'}, {title: '9'}],
    [
        {title: 'Undo', style: {fontSize: 25}},
        {title: '0'},
        {title: 'Submit', style: {fontSize: 25}}
    ],
]

let mapDispatchToProps = dispatch => ({
    actions: bindActionCreators({
        onScoreChange,
        onScoreSubmit,
        onScoreClear,
        onScoreUndo
    }, dispatch)
})

export const KeyBoard = ({actions}) => {
    let onButtonPress = title => {
        if (title === 'Undo') {
            return actions.onScoreUndo()
        }

        if (title === 'Submit') {
            actions.onScoreSubmit()
            return actions.onScoreClear()
        }

        actions.onScoreChange(title)
    }

    return (
        <View style={{alignItems: 'center', flexDirection: 'column'}}>
            {buttons.map((row, i) => (
                <View key={i} style={{alignItems: 'center', flexDirection: 'row'}}>
                    {row.map(button => (
                        <TouchableOpacity
                            key={button.title}
                            style={styles.button}
                            onPress={onButtonPress.bind(this, button.title)}
                        >
                            <Text style={[styles.buttonText, button.style]}>
                                {button.title}
                            </Text>
                        </TouchableOpacity>
                    ))}
                </View>
            ))}
        </View>
    )
}

const styles = StyleSheet.create({
    button: {
        backgroundColor: 'white',
        marginRight: 5,
        marginBottom: 5,
        height: 100,
        width: 100,
        justifyContent: 'center',
        alignItems: 'center',
        alignSelf: 'stretch',
    },
    buttonText: {
        fontSize: 55,
        fontWeight: '500',
        color: '#222'
    }
})

export default connect(null, mapDispatchToProps)(KeyBoard)
