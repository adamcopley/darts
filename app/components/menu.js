import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import {
    StyleSheet,
    Text,
    View,
    Button,
    TextInput,
    Picker
} from 'react-native'

import {
    onGameSetName,
    onGameStart,
    onGameSetTotal
} from '../actions'

const mapStateToProps = state => ({
    total: String(state.total),
    p1: state.p1.name,
    p2: state.p2.name
})

const mapDispatchToProps = dispatch => bindActionCreators({
    onGameSetName,
    onGameStart,
    onGameSetTotal
}, dispatch)

const Menu = ({
    p1,
    p2,
    total,
    setName,
    onGameSetTotal,
    onGameSetName,
    onGameStart
}) => (
    <View style={styles.container}>
        <Text style={styles.title}>{p1} vs {p2}</Text>
        <View>
            <Text style={styles.inputTitle}>Player 1:</Text>
            <TextInput
                style={styles.input}
                placeholder={p1}
                onChangeText={name => onGameSetName('p1', name)}
            />
        </View>
        <View>
            <Text style={styles.inputTitle}>Player 2:</Text>
            <TextInput
                style={styles.input}
                placeholder={p2}
                onChangeText={name => onGameSetName('p2', name)}
            />
        </View>
        <View style={[styles.container, {flex: 0 }]}>
            <Text style={styles.inputTitle}>Total Score:</Text>
            <Picker
                enabled
                style={styles.picker}
                selectedValue={total}
                onValueChange={total => onGameSetTotal(total)}>
                <Picker.Item label="101" value="101" />
                <Picker.Item label="301" value="301" />
                <Picker.Item label="501" value="501" />
            </Picker>
        </View>
        <Button
            onPress={onGameStart}
            title="Start game"
        />
    </View>
)

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F5FCFF',
    },
    title: {
        color: 'black',
        fontSize: 40,
        marginBottom: 20
    },
    inputTitle: {
        color: 'black',
        fontSize: 30,
        marginTop: 10
    },
    input: {
        height: 25
    },
    picker: {
        width: 90,
    }
})

export default connect(mapStateToProps, mapDispatchToProps)(Menu)