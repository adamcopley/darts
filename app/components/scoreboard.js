import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
    StyleSheet,
    Text,
    View,
} from 'react-native'

import getHistory from '../helpers/getHistory'

const mapDispatchToProps = state => ({
    history: state.history,
    currentPlayer: state.currentPlayer,
    p1: state.p1,
    p2: state.p2
})

export const ScoreBoard = ({p1, p2, currentPlayer, history}) => (
    <View style={styles.wrapper}>
        <View>
            <Text style={styles.title}>
                {p1.name}
            </Text>
            <Text style={[styles.score, {color: currentPlayer !== 'p1' ? 'grey' : 'black'}]}>
                {p1.score}
            </Text>
            <Text>
                {history.length > 0 && getHistory(history, 1)}
            </Text>
        </View>
        <View>
            <Text style={styles.title}>
                {p2.name}
            </Text>
            <Text style={[styles.score, {color: currentPlayer !== 'p2' ? 'grey' : 'black'}]}>
                {p2.score}
            </Text>
            <Text>
                {history.length > 0 && getHistory(history, 2)}
            </Text>
        </View>
    </View>
)

const styles = StyleSheet.create({
    wrapper: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-around',
        maxHeight: 150,
        width: '100%',
    },
    title: {
        fontSize: 25,
    },
    score: {
        fontSize: 70
    }
})

export default connect(mapDispatchToProps)(ScoreBoard)
