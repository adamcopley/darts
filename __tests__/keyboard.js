import 'react-native'
import React from 'react'
import { KeyBoard } from '../app/components/keyboard'

import renderer from 'react-test-renderer'

test('KeyBoard renders correctly', () => {
    let tree = renderer.create(
        <KeyBoard actions={{}} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
})