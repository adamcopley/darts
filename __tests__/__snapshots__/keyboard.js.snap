// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyBoard renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        2
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        3
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        4
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        5
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        6
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        7
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        8
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        9
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            Object {
              "fontSize": 25,
            },
          ]
        }
      >
        Undo
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            undefined,
          ]
        }
      >
        0
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginRight": 5,
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 55,
              "fontWeight": "500",
            },
            Object {
              "fontSize": 25,
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
